src/NumericalAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -423,14 +423,55 @@ void add_received_variables(
     const TemporalId& temporal_id) noexcept {
   db::mutate<Tags::IndicesOfFilledInterpPoints<TemporalId>,
              Tags::InterpolatedVars<InterpolationTargetTag, TemporalId>>(
-      box, [&temporal_id, &vars_src, &global_offsets ](
+      box, [&temporal_id, &vars_src, &global_offsets](
                const gsl::not_null<
                    std::unordered_map<TemporalId, std::unordered_set<size_t>>*>
                    indices_of_filled,
                const gsl::not_null<std::unordered_map<
                    TemporalId, Variables<typename InterpolationTargetTag::
                                              vars_to_interpolate_to_target>>*>
                    vars_dest_all_times) noexcept {
+        if (UNLIKELY(0 == vars_dest_all_times->count(temporal_id))) {
```

Comment:
When testing a bit more last night, I found that we need another similar check in `have_data_at_all_points`, because after the control flow exits this routine early, it will try to check if it needs to clean up the already cleaned-up temporal id.